### PR TITLE
Michael

### PR DESCRIPTION
**Pull request by bgraham89**
I've made this as a pull request so I can add comments to it

### CODE REVIEW

File: classes.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        self.name = name
+        self.cost = cost
+        self.design_weeks = design_weeks  # Weeks needed to manufacture the component
+
+class StaffMember:
+    def __init__(self, title, cost_per_day, employment_type, weeks_assigned):
+        self.title = title  # Job Title
+        self.cost_per_day = cost_per_day
+        self.employment_type = employment_type  # In-House or Agency
+        self.weeks_assigned = weeks_assigned  # Amount of weeks assigned to the job
```
Comment (bgraham89) [inline classes.py]: We could get the estimator to automatically assign tasks to members if we did the following things:

1. Assign each staff member a maximum amount of days they can work. For example, we know the project length will be ~12 months or 52 weeks and each and most staff will work 5 days a week, so those staff could have a `workday_cap` of 52 x 5. The advantage of that would be that our programme can do the allocation of tasks for us by selecting staff for tasks from a pool of staff as long as those staff havent met their workday_cap amount of days. 
2. We could record staff member capabilities as strings in a set. So our StaffMember, constructor might look like this instead:   `def __init__(self, title, cost_per_day, employment_type, workday_cap, capabilities): `. The application could use this information as a constraint when assigning tasks.
3. Update the total_design_cost method of the ProjectEstimator class to allocate each week to a suitable member of staff, and sum the result (while tracking how many days each person has available, perhaps in a 'counter' object)

Comment (michaelsammueller) [reply classes.py]: That's a great idea, I'll get right to it! Thanks for having a look @bgraham89 !

**Comment by michaelsammueller** [comment]
I've updated the `classes.py` file according to @bgraham89 's comments.

- Every `StaffMember` instance is now stored as an object within the `ProjectEstimator` class. This allows us to access their attributes later.
- Every `StaffMember` now has a `workday_cap`, `capabilities`, and `workdays_used` attribute. This allows us to set an overall cap on their work days, their capabilities (as part of a set), and use `workdays_used` as a counter to check how many days that staff member has already worked.
- Every `HardwareComponent` and `SoftwareComponent` now has a `required_capabilities` attribute (stored as a set), which we can use to define the required capabilities for said component (f.e. "Hardware Design", "Software Design", "Hardware Manufacturing", or whatever else we want to define).
- The `total_design_cost()` and `total_manufacturing_cost()` methods iterate over each component to check their required capabilities. They then iterate over every staff member in the `project_staff` dictionary and check their capabilities. If a staff member has the required capability for the current component, then the staff member will be assigned to that component - as long as the required time does not exceed his/her work days.

Please do let me know if you see anything obvious that I have missed.

**Comment by michaelsammueller** [comment]
Fixed an inconsistency where both "days" and "weeks" were used to measure time. Now only "days" are used to keep it consistent with cost calculation.

**Pull request merged**